src/tests/arrow_func_tests.rs

```diff
@@ -0,0 +1,129 @@
+                .transform_status
+                .is_some_and(|status| status.status == Status::NotModified),
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_arrow_block_not_modified() -> Result<(), String> {
+        let original_code = "{const a = (arg) => { return arg ? true : false }}".to_string();
+        let js_file = "test.js".to_string();
+        let rewritten = rewrite_js(original_code, js_file).map_err(|e| e.to_string())?;
+
+        assert_that(
+            &rewritten
+                .transform_status
+                .is_some_and(|status| status.status == Status::NotModified),
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_arrow_object_not_modified() -> Result<(), String> {
+        let original_code = "{const a = (arg) => ({ test: true }) }".to_string();
+        let js_file = "test.js".to_string();
+        let rewritten = rewrite_js(original_code, js_file).map_err(|e| e.to_string())?;
+
+        assert_that(
+            &rewritten
+                .transform_status
+                .is_some_and(|status| status.status == Status::NotModified),
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_arrow_object_2_not_modified() -> Result<(), String> {
+        let original_code = "{const a = (arg) => { test: true } }".to_string();
+        let js_file = "test.js".to_string();
+        let rewritten = rewrite_js(original_code, js_file).map_err(|e| e.to_string())?;
+
+        assert_that(
+            &rewritten
+                .transform_status
+                .is_some_and(|status| status.status == Status::NotModified),
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_arrow_modified() -> Result<(), String> {
+        let original_code = "{const a = (arg) => arg ? `hello ${arg}` : '' }".to_string();
+        let js_file = "test.js".to_string();
+        let rewritten = rewrite_js(original_code, js_file).map_err(|e| e.to_string())?;
+
+        assert_that(
+            &rewritten
+                .transform_status
+                .is_some_and(|status| status.status == Status::Modified),
+        );
+        assert_that(&rewritten.code).contains("const a = (arg)=>{
+        let __datadog_test_0;
+        return arg ? (__datadog_test_0 = arg, _ddiast.tplOperator(`hello ${__datadog_test_0}`, __datadog_test_0)) : '';
+    };");
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_arrow_with_block_modified() -> Result<(), String> {
+        let original_code = "{
+          const a = (arg) => { return arg ? `hello ${arg}` : '' }
+          const b = `${a('world')} bye`
+        }"
+        .to_string();
+        let js_file = "test.js".to_string();
+        let rewritten = rewrite_js(original_code, js_file).map_err(|e| e.to_string())?;
+
+        assert_that(
+            &rewritten
+                .transform_status
+                .is_some_and(|status| status.status == Status::Modified),
+        );
+        assert_that(&rewritten.code).contains("const a = (arg)=>{
```

Comment:
`let let __datadog_test_0` should be defined twice, in and out of lambda, without it, the one that is used in `const b` is not defined. 
```suggestion
        assert_that(&rewritten.code).contains("let __datadog_test_0
const a = (arg)=>{

Reply:
i wanted to test that __datadog_test_0 was defined inside the arrow but I changed the assertion to include also __datadog_test_0 from outside